### PR TITLE
 Baseline frontend-backend connection #207 

### PR DESCRIPTION
**Pull request by momanyisamuel**
### Contributor checklist

- [x] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch

---

### Related issue

- #207

### CODE REVIEW

**Comment by andrewtavis** [comment]
@wkyoshida, post call we're still getting some issues with Docker not working for the database. I'm going to look into a few example projects to see if there are some suggestions for this, but would be great if you can see some errors in the `docker-compose.yml` from the get go :)

File: docker-compose.yml
Comment (wkyoshida) [inline docker-compose.yml:12]: I see that there is a `backend/backend/.env.example` file that was created. Should that be what is referenced here? Or should the file be renamed to `backend/backend/.env` instead?

File: docker-compose.yml
Comment (wkyoshida) [inline docker-compose.yml:33]: Should `DATABASE_PORT` rather be `5432`? Or was there a reason to choose `3306`?

File: backend/apps/views/__init__.py
Comment (wkyoshida) [inline backend/apps/views/__init__.py:1]: A more general comment, but I would see to avoid omitting a newline character at the end of files. Just always add one.

This is more due to some tools having issues processing files, since some of them rely on a newline character to determine the end of the file. Does not always happen, but it can, which is why GitHub usually adds this warning.

**Review by wkyoshida — COMMENTED** [review]
Not a thorough review, but just a couple of items that jumped out at me at first glance
:rocket:

File: docker-compose.yml
Comment (andrewtavis) [reply docker-compose.yml:12]: I've seen @momanyisamuel's convention other places as well – that there's an `.env.example` file that people would need to convert to their own `.env` file. Would you have another suggestion on this? I guess for dev purposes we can share an admin pass and there needs to be another one that's hidden what's live?

**Comment by andrewtavis** [comment]
> Not a thorough review, but just a couple of items that jumped out at me at first glance 🚀

Ya no stress, I can do the full review!

File: backend/apps/views/__init__.py
Comment (andrewtavis) [reply backend/apps/views/__init__.py:1]: Ya I think at one point we need to do a sit down for the dev community that's focussed on what tools we're using in our environment. Specifically we all should be using the same formatting process, and this process should enforce having a line at the end of the file on save. TBH I have not consciously done this in years, and if I didn't have an extension doing it for me I probably would forget :) What it is that we agree on can then be added to [STYLEGUIDE.md](https://github.com/activist-org/activist/blob/main/STYLEGUIDE.md) as well as [.vscode/extensions.json](https://github.com/activist-org/activist/blob/main/.vscode/extensions.json). This wouldn't be a full dev sync, but maybe 15 min in one where people are told to prepare before. Maybe the one after next as the next one will be quite busy with backend again?

For the above an idea on reinforcing this would be to have another workflow similar to the [maintainer checklist](https://github.com/activist-org/activist/pull/209#issuecomment-1620588231) that would fire on first PRs to the repo. We could shift the check of the commit messages to this as it really just needs to be done once, and beyond that we could have a point that directs people to the style guide and links to the extensions that we as a community think they should be using.

Plus we can all take a look at @momanyisamuel's VS Code setup and marvel at his sidebar on the left 🤪😅😊

File: docker-compose.yml
Comment (momanyisamuel) [reply docker-compose.yml:12]: Yeah the `.env.example` can just be renamed to `.env`

File: docker-compose.yml
Comment (momanyisamuel) [reply docker-compose.yml:33]: Actually this correct will make the changes

File: docker-compose.yml
Comment (andrewtavis) [reply docker-compose.yml:12]: We'd need to remove `.env` in the [.gitignore](https://github.com/activist-org/activist/blob/main/.gitignore) then :)

Comment (andrewtavis) [reply docker-compose.yml:12]: And how would we hide the actual values later on when we need them?

File: backend/backend/.env.example
Comment (andrewtavis) [inline backend/backend/.env.example:1]: @momanyisamuel, we still need to delete the one that's here, right?

**Comment by andrewtavis** [comment]
I found the following two repos that might serve as a good reference for some of the settings files, btw:
- https://github.com/melehin/django-celery-vue-template
- https://github.com/MakSzafa/docker-django-vue-postgres-project-template

File: docker-compose.yml
Comment (wkyoshida) [reply docker-compose.yml:12]: > I guess for dev purposes we can share an admin pass and there needs to be another one that's hidden what's live?

Yup - I would think that, with this dev instance, people would really only run it locally. So the credentials would be "shared" in the sense that it would just be the same one for anyone spinning it up and running it locally. 

> And how would we hide the actual values later on when we need them?

That would depend on the production deployment strategy. Deploying within containers, [Docker Compose secrets](https://docs.docker.com/compose/use-secrets/) can be used. Where to store them beforehand and how to fetch them though depends on the deployment strategy, e.g. AWS has a [Secrets Manager](https://aws.amazon.com/secrets-manager/). Seems like Netlify has one [alternative](https://www.netlify.com/blog/a-guide-to-storing-api-keys-securely-with-environment-variables/). GitHub has their own [secrets manager](https://docs.github.com/en/actions/security-guides/encrypted-secrets), so they could be used potentially within GitHub Actions to fetch the secrets, build the production containers, store the secrets inside the containers, and then deploy the containers.

i.e. TBD, but there are various options. Important thing though, as you're likely alluding to, is to not expose them as regular environment variables.

File: backend/backend/.env.example
Comment (momanyisamuel) [reply backend/backend/.env.example:1]: Yes this will be removed as discussed

**Comment by momanyisamuel** [comment]
Hi @andrewtavis please review and see if this is good to merge

File: docker-compose.yml
Comment (Vince155) [reply docker-compose.yml]: I think another option for storing secrets would be to encrypt them using something like [Mozilla SOPS](https://github.com/getsops/sops#encrypting-an-existing-file) that way a file like `.env.example` can be encrypted as `.enc.env.example`.

**Comment by andrewtavis** [comment]
Sorry for this taking so long, @momanyisamuel. There's a lot I still need to do to get settled into the new place + work + etc. I'll review this this evening :)

File: backend/apps/models.py
Comment (andrewtavis) [inline backend/apps/models.py:13]: `verification_partner` :)

File: backend/apps/models.py
Comment (andrewtavis) [inline backend/apps/models.py:105]: Could you let me know why we're doing `1./2. CAP_VERBOSE_NAME` for the `verbose_name_plural`, @momanyisamuel? Is this something to format it for the admin panel?

File: backend/apps/models.py
Comment (andrewtavis) [inline backend/apps/models.py:279]: `Organization` without the `Z` :)

File: backend/apps/urls.py
Comment (andrewtavis) [inline backend/apps/urls.py:5]: `app_name` doesn't appear to have been used?

File: backend/apps/views/__init__.py
Comment (andrewtavis) [inline backend/apps/views/__init__.py:1]: For Django do we have to do explicit imports within `__init__.py`? For packages I read a while ago that this should not be done.

File: backend/apps/urls.py
Comment (andrewtavis) [reply backend/apps/urls.py:5]: Or is this a variable that's set here but read elsewhere? I'm seeing that we have `apps.User` and `apps.url`.

File: docker-compose.yml
Comment (andrewtavis) [inline docker-compose.yml:29]: Just out of curiosity, why does the frontend port need to be switched to `4000`? If this change does go through, we need to make sure to update the readme and the contributing guidelines :)

**Comment by andrewtavis** [comment]
General review's all done, @momanyisamuel :) Mostly is questions for the implementation and some minor minor changes being needed. Happy to merge once these are answered/fixed and then we can continue the conversation in #174 😊

File: docker-compose.yml
Comment (andrewtavis) [reply docker-compose.yml]: With the decision to leverage supabase I think checking out their [secrets management](https://supabase.com/docs/guides/functions/secrets) would also be good :) I'll look into this more over the weekend after this is merged 😊

File: backend/apps/models.py
Comment (momanyisamuel) [reply backend/apps/models.py:105]: yes this is for the admin panel

File: backend/apps/views/__init__.py
Comment (momanyisamuel) [reply backend/apps/views/__init__.py:1]: Well i think this will probably change with the use of Django REST framework

File: docker-compose.yml
Comment (momanyisamuel) [reply docker-compose.yml:29]: will change this to 3000

**Comment by momanyisamuel** [comment]
Made changes as reviewed @andrewtavis please let me know if the changes are okay

File: backend/apps/views/__init__.py
Comment (andrewtavis) [reply backend/apps/views/__init__.py:1]: Totally fine then :) Just checking based on prior experiences and the love of not having import statements in Nuxt ❤️🙃

**Review by andrewtavis — APPROVED** [review]
We've got ourselves a backend!! 😊😊🚀 Thanks so much for the hard work and patience on this, @momanyisamuel! I'll have some time on Saturday to work on this a bit more. Would be great if you could sketch some future steps in #174 :)

**Comment by momanyisamuel** [comment]
Great glad this is done @andrewtavis

**Pull request merged**
